crates/ibc/src/core/ics02_client/handler/upgrade_client.rs

```diff
@@ -24,13 +21,6 @@ where
 
     ctx.validate_message_signer(&signer)?;
 
-    // Temporary has been disabled until we have a better understanding of some design implications
```

Comment:
I can't find where ibc-go looks if the `old_client_state` is expired; can you point me to it? 

(line 49, GH won't let me put a comment there since it wasn't modified in this PR)

Reply:
ofc, [Here](https://github.com/cosmos/ibc-go/blob/c3b12b11f1f60e0c7ba45ccc4c0cd8ae9365e538/modules/core/02-client/keeper/client.go#L125-L127) upon calling the `GetClientStatus` method, it goes through the [status check](https://github.com/cosmos/ibc-go/blob/d840c699d988e32cb55f2c289d69f8ef2e5cea50/modules/core/02-client/keeper/keeper.go#L414)

Reply:
Ah right. The "client expired" check is a [tendermint one though](https://github.com/cosmos/ibc-go/blob/295bab01f633633e0d6136ba9b974c0e248edbe5/modules/light-clients/07-tendermint/client_state.go#L97-L98), so we should have it in the tendermint client instead

Reply:
I see your point. Just note that, with the current design, having `expired` and even `confirm_not_frozen` as client-wide interfaces, I suggest we mend this while going through #536 and wait for changes from #296
Particularly given that, the `expired` needs a `host_timestamp` from context for calculation, which is not available within the `verify_upgrade_client`

Reply:
Makes sense